vembrane/representations.py

```diff
@@ -44,116 +27,117 @@ def get(self, item, default=NA):
-        ann_key: str,
-    ):
-        self._record_idx = record_idx
-        self._record = record
-        self._record_info = record.info
-        self._header_info_fields = header_info_fields
-        self._ann_key = ann_key
-        self._info_dict: Dict[str, NvIntFloatStr] = {}
-
-    def __getitem__(self, item):
-        try:
-            return self._info_dict[item]
-        except KeyError:
-            if item == "END":
-                # pysam removes END from info. In order to fit with user expectations,
-                # (they will expect INFO["END"] to work) we emulate it being present by
-                # inferring it from pysams record representation.
-                value = get_end(self._record)
-            else:
-                try:
-                    if item == self._ann_key:
-                        raise KeyError(item)
-                    untyped_value = self._record_info[item]
-                except KeyError:
-                    if item in self._header_info_fields:
-                        value = NA
-                    else:
-                        raise UnknownInfoField(self._record_idx, self._record, item)
-                else:
-                    value = type_info(
-                        untyped_value,
-                        self._header_info_fields[item],
-                        item,
-                        self._record_idx,
-                    )
-            self._info_dict[item] = value
-            return value
+# class Format(NoValueDict, DefaultGet):
+#     def __init__(
+#         self,
+#         record_idx: int,
+#         record: VCFRecord,
+#         name: str,
+#         number: str,
+#         # record_samples: VCFRecordSamples,
+#         record_format: VCFRecordFormat,
+#     ):
+#         self._record_idx = record_idx
+#         self._record = record
+#         self._name = name
+#         self._number = number
+#         # self._record_samples = record_samples
+#         self._record_format = record_format
+#         self._sample_values: Dict[str, NvIntFloatStr] = {}
+
+#     def __getitem__(self, sample):
+#         try:
+#             return self._sample_values[sample]
+#         except KeyError:
+#             try:
+#                 record_sample = self._record_format[self._name][sample]
+#             except KeyError:
+#                 raise UnknownSample(self._record_idx, self._record, sample)
+#             value = type_info(record_sample, self._number,
+# self._name, self._record_idx)
+#             self._sample_values[sample] = value
+#             return value
+
+
+# class Formats(NoValueDict):
+#     def __init__(
+#         self,
+#         record_idx: int,
+#         record: VCFRecord,
+#         header_format_fields: Dict[str, str],
+#     ):
+#         self._record = record
+#         self._record_idx = record_idx
+#         self._header_format_fields = header_format_fields
+#         self._record_format = record.format
+
+#     @cache
```

Comment:
The `__getitem__` method has a `@cache` decorator, not sure if that was actually beneficial or not and if we should keep in the new backend specific formats classes?

Reply:
I added the @cache decorator, because the old code was actually a self implemented cache. I now add the cache decorator where I think it might be helpful for some queries